packages/eslint-plugin/docs/rules/no-duplicate-type-constituents.md

```diff
@@ -57,3 +57,7 @@ When set to true, duplicate checks on intersection type constituents are ignored
 ### `ignoreUnions`
 
 When set to true, duplicate checks on union type constituents are ignored.
+
+## When Not To Use It
+
+If you don't mind excess types taking up space in intersections or unions, you can avoid this rule.
```

Comment:
This is the sort of sentence I was meaning by my comment.
All this really does is invert the rule name.

"If you don't want your types to have no duplicate type constituents, use this rule"
"If you don't mind your types having duplicate type constituents, don't use this rule"

Personally I don't think this one adds any value to the rule docs.
Which is why I say "prefer to add one, but not enforce they all have one".

Reply:
Finally getting back to this! I think I got this intent in all the rules I could. For any others I added a `<!-- Intentional omission: When Not To Use It -->` comment. WDYT?

Reply:
Yeah I'm more than happy if we have a way to explicitly opt-out!